aten/src/ATen/WrapDimUtils.h

```diff
@@ -7,6 +7,10 @@
 namespace at {
 
 static inline int64_t maybe_wrap_dim(int64_t dim, int64_t dim_post_expr, bool wrap_scalar=true) {
+  // dim_post_expr can be negative only when wrap_scalar is true, which will
+  // automatically set it to 1.
```

Comment:
Err I don't know what it means for `dim_post_expr` to be negative. That just sounds like a bug in the code.

I think what we want to say is "if dim_post_expr is 0 and wrap_scalar, then dim must be in the range [-1, 0]. This is a weird special case for scalar tensors and manifests in e.g. torch.sum(scalar_tensor, 0)"